### ISSUE TITLE
Announcing IISConfigurator v0.1 (replacement for Status Monitor)

### ISSUE THREAD

**Issue by MS-TimothyMothra** [Microsoft/ApplicationInsights-Announcements#21]
Hello,
I’ve just published v0.1-alpha of IISConfigurator. This is our replacement for StatusMonitor.

I’ve organized all our docs for the IISConfigurator here:  https://github.com/Microsoft/ApplicationInsights-Home/blob/master/Samples/IISConfigurator/TOC.md
This includes:
- A Quick Start Guide
- Detailed getting started instructions
- API reference for the new PowerShell Module
- A page 

dedicated to Troubleshooting and Known Issues

Disclaimer: This is a prototype application. We do not recommend using this on your production environments.
Please pay close attention to the known issues.
Your apps may have issues if they include any of the following DLLs in the bin directory:
- Microsoft.ApplicationInsights.dll
- Microsoft.AspNet.TelemetryCorrelation.dll
- System.Diagnostics.DiagnosticSource.dll
 

I am very interested in your feedback!
Please don’t hesitate to contact me if you have any issues or comments.
Specifically,
- I would love to hear about your experiences getting started
- And what you would like to see moving forward

**Comment by tsahi** [comment]
Why PowerShell? Would there be a GUI for this?

**Comment by bogdangrigg** [comment]
This sounds potentially very useful - when would GA be planned?
Also, are you targeting feature parity with the embedded SDK option?

**Comment by jawn** [comment]
> We expect our first Stable release sometime in May 2019.

From https://github.com/Microsoft/ApplicationInsights-Home/blob/master/Samples/IISConfigurator/TOC.md

**Comment by MS-TimothyMothra** [comment]
@tsahi We don't currently have plans for a new GUI but we will listen to customer's needs. If there's a specific feature of the GUI that you depend on, please share your use case so we can take that into consideration. :) Right now our focus is delivering a fully functional PowerShell API. This will address the automation needs of our largest enterprise customers.

@bogdangrigg Sometime in May 2019. :)  This product uses the [.NET SDK](https://github.com/Microsoft/ApplicationInsights-dotnet) so you would receive the features and bug/performance fixes of this SDK.

**Comment by jmoore4xrx** [comment]
Thanks.   I am not looking for GUI per se.   Just found it odd that the custom Log I created did not show in current Advanced Settings > Data > Custom Logs list.

I’d prefer the PowerShell API.  Is there an outlook for this?  Can I sing up for  notification of release?

Thanks again,
JAM/John A. Moore

**Comment by MS-TimothyMothra** [comment]
@jmoore4xrx  I will continue to make announcements on this repo.

**Comment by tsahi** [comment]
@MS-TimothyMothra we have only a few servers we maintain, so we don't use automation for deployment of App Insights. In this scenario, it's easier to use a GUI than memorize console commands.

**Comment by MS-TimothyMothra** [comment]
@tsahi Thanks for the information. How often do you re-deploy Application Insights to your few servers?

The former Status Monitor would instrument individual applications by dropping DLLs into the bin directory. This required a re-install every time you re-deploy your original application.

The newer IIS Configurator will install once on your server to instrument IIS. Your applications will light up even after re-deploying. I hope that this newer version would create less maintenance for you overall. :)

Your feedback here is super valuable. Thank you!

**Comment by jawn** [comment]
> The newer IIS Configurator will install once on your server to instrument IIS. Your applications will light up even after re-deploying. I hope that this newer version would create less maintenance for you overall. 

Thanks, that's a super improvement to support CI/CD scenarios without headaches 👍.

**Comment by jawn** [comment]
Question: the docs say
> This is a prototype application. We do not recommend using this on your production environments.

Will that disclaimer be removed once it is stable?

**Comment by MS-TimothyMothra** [comment]
@jawn Yes, we will remove that disclaimer with our first stable release.
Part of our requirements for a stable release is a security review.

We are confident that the code is safe because this code base is shared with some Azure Extensions that are already live in production. This difference here is because of the on-prem scenario and we can't control the security of the installation folder. The next best thing is to review that we're not exposing you to harm.

**Comment by tsahi** [comment]
@MS-TimothyMothra that's not the experience we've had. We installed Status Monitor once, enabled it for the applications we needed on the machine, and it just works. We integrated the App Insights SDK into our application, and write logs from our code. We added status monitor because of [this page](https://docs.microsoft.com/en-us/azure/azure-monitor/app/monitor-performance-live-website-now) (or a similar one), to get the extra data the SDK doesn't provide. I think the note above the table didn't exist at the time. We do get detailed dependency information though.

**Comment by MS-TimothyMothra** [comment]
@tsahi. Thanks for sharing, that's perfect.
In the background, StatusMonitor was installing an extension called **Instrumentation Engine.**
In the v0.2 update, we're introducing an api to install only that Instrumentation Engine and nothing else. This standalone installation wasn't available in Status Monitor.
